src/noise.jl

```diff
@@ -153,7 +154,8 @@ function simulate_noise(wf::RDWaveform, preamp::PreAmp)
     # wf values are in eV (without u"eV" units attached), noise sigma is in keV
     # if we're not simulating noise from scratch, we'll do this anyway, but noise sigma will be 0
     # -> maybe there's a better way?
-    noise_σ = ustrip(uconvert(u"eV", preamp.noise_σ))
+    noise_σ = preamp.noise_σ_keV == 0 ? preamp.noise_σ_ADC / preamp.gain : uconvert(u"eV", preamp.noise_σ_keV)
+    noise_σ = ustrip(noise_σ)
```

Comment:
Just to be safe that `noise_σ` is always in units of `eV`, you could do `ustrip(u"eV", noise_σ)` here.

Reply:
Thanks, added this to be sure, ran tests with noise in ADC input, works correctly.

Reply:
It would be good to add the tests you're talking about also to `test/runtest.jl`, such that they get checked with every modification of the source code in the future!